src/kiota/Extension/KiotaHostExtensions.cs

```diff
@@ -94,4 +100,40 @@ static void ConfigureServiceContainer(HostBuilderContext context, IServiceCollec
             return OperatingSystem.IsFreeBSD() ? "freebsd" : null;
         }
     }
+
+    internal static string AcquisitionChannel(string? path)
+    {
+        // Docker
+        if (Environment.GetEnvironmentVariable("KIOTA_CONTAINER") == "true") return "docker";
+        if (path != null && !string.IsNullOrWhiteSpace(path))
+        {
+
+            var absolutePath = Path.GetFullPath(path);
+            var homeDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+            // Dotnet tool
+            if (absolutePath.StartsWith(Path.Join(homeDir, ".dotnet", "tools")))
+            {
+                return "dotnet_tool";
+            }
+            // ASDF
+            if (absolutePath.StartsWith(Path.Join(homeDir, ".asdf")))
+            {
+                return "asdf";
+            }
+            // Extension
+            if (absolutePath.Contains(".vscode"))
+            {
+                return "extension";
+            }
+
+#if MACOS
```

Comment:
I believe it's fairly common to use homebrew on linux as well. What do you think about removing the conditional directives?

Reply:
Is the path the same? I can add a linux conditional. This code path doesn't make sense on windows.